services/rsources/stats_collector.go

```diff
@@ -286,3 +272,75 @@ func (r *statsCollector) buildStats(jobs []*jobsdb.JobT, failedJobs map[uuid.UUI
 		}
 	}
 }
+
+type Identifier struct {
+	JobRunID string
+	RecordID string
+	Target   JobTargetKey
+}
+
+type GetJobIdentifiersFunc func(jp json.RawMessage) Identifier
```

Comment:
Let's take advantage of the language's ability to return multiple values and avoid having to declare yet another struct
```suggestion
type parametersParser func(jp json.RawMessage) (jobRunID string, recordID string, target JobTargetKey)
```